lib/std/crypto/scrypt/mcf_encoding.zig

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2021 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+
+const std = @import("std");
+const crypto = std.crypto;
+const math = std.math;
+const meta = std.meta;
+const mem = std.mem;
+
+const scrypt = @import("scrypt.zig");
+
+pub const Error = error{
+    ParseError,
+    InvalidAlgorithm,
+};
+
+pub const McfEncoding = struct {
+    const Self = @This();
+    const map64 = "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+
+    fn encodedLen(comptime len: usize) usize {
+        return (len * 4 + 2) / 3;
+    }
+
+    fn intEncode(dst: []u8, src: anytype) void {
+        var n = src;
+        for (dst) |*x, i| {
+            x.* = map64[@truncate(u6, n)];
+            n = math.shr(@TypeOf(src), n, 6);
+        }
+    }
+
+    fn sliceEncode(comptime len: usize, dst: *[encodedLen(len)]u8, src: [len]u8) void {
+        var i: usize = 0;
+        while (i < src.len / 3) : (i += 1) {
+            intEncode(dst[i * 4 ..][0..4], mem.readIntSliceLittle(u24, src[i * 3 ..]));
+        }
+        const leftover = src[i * 3 ..];
+        var v: u24 = 0;
+        for (leftover) |x, j| {
+            v |= @as(u24, x) << @intCast(u5, j * 8);
+        }
+        intEncode(dst[i * 4 ..], v);
+    }
+
+    fn intDecode(comptime T: type, src: *const [(meta.bitCount(T) + 5) / 6]u8) !T {
```

Comment:
Why does this take a pointer for `src` while `sliceEncode` takes an array as-is?